### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kolesroma**
megre to main

### CODE REVIEW

File: src/main/java/com/epam/emotionalhelp/controller/QuestionController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.Collection;
+
+@RestController
+public class QuestionController {
+    private final QuestionService questionService;
+
+    public QuestionController(QuestionService questionService) {
+        this.questionService = questionService;
+    }
```
Comment (kolesroma) [inline src/main/java/com/epam/emotionalhelp/controller/QuestionController.java:16]: @RequiredArgsConstructror

File: src/main/java/com/epam/emotionalhelp/controller/QuestionController.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.epam.emotionalhelp.controller;
+
+import com.epam.emotionalhelp.module.Question;
+import com.epam.emotionalhelp.service.QuestionService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collection;
+
```
Comment (kolesroma) [inline src/main/java/com/epam/emotionalhelp/controller/QuestionController.java:9]: @RequestMapping /questions

File: src/main/java/com/epam/emotionalhelp/module/Question.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.epam.emotionalhelp.module;
+
+public class Question {
```
Comment (GeloPakDev) [inline src/main/java/com/epam/emotionalhelp/module/Question.java:3]: Add Lombok library

File: src/main/java/com/epam/emotionalhelp/module/Question.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    private Integer id;
```
Comment (GeloPakDev) [inline src/main/java/com/epam/emotionalhelp/module/Question.java:4]: Use Long for id

File: src/main/java/com/epam/emotionalhelp/service/impl/QuestionServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Service
+public class QuestionServiceImpl implements QuestionService {
+    @Override
+    public Collection<Question> getAll() {
```
Comment (kolesroma) [inline src/main/java/com/epam/emotionalhelp/service/impl/QuestionServiceImpl.java:14]: /findAll - convnsion

**Review by maximelyashevich — CHANGES_REQUESTED** [review]
Hi @kolesroma,
Please review the above comments and refactor your code.

**Comment by kolesroma** [comment]
Megre pull request #22 is failed because I didn't commit pom.xml file to avoid merge conflict with Megre pull request #18

File: src/main/java/com/epam/emotionalhelp/controller/QuestionController.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import com.epam.emotionalhelp.service.QuestionService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collection;
+
+@RestController
+@RequestMapping("/questions")
```
Comment (maximelyashevich) [inline src/main/java/com/epam/emotionalhelp/controller/QuestionController.java:13]:
```suggestion
@RequestMapping("/v1/questions")
```
Let's also add versioning for REST API.

File: src/main/java/com/epam/emotionalhelp/controller/QuestionController.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+@RequiredArgsConstructor
+public class QuestionController {
+    private final QuestionService questionService;
+
+    @GetMapping("/findAll")
```
Comment (maximelyashevich) [inline src/main/java/com/epam/emotionalhelp/controller/QuestionController.java:18]:
```suggestion
    @GetMapping
```
No need to use `/findAll` - we can request just `/v1/questions` to get all the questions.

File: pom.xml [hunk 8]
```diff
@@ -3,24 +3,77 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <version>${spring.boot.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <!--MYSQL-->
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.30</version>
```
Comment (maximelyashevich) [inline pom.xml:46]: Please also add `mysql.version`, `junit.version`, `h2.version`, `lombok.version` properties.

**Review by maximelyashevich — COMMENTED** [review]
Hi @kolesroma,
Please provide a small refactoring and close this PR.

Regards,
Maks

**Review by maximelyashevich — APPROVED** [review]
Hi @kolesroma,
Good job - and thanks for the refactoring! Please close this PR.

Regards,
Maks

**Pull request merged**